implementations/rust/ockam/ockam_identity/src/credential.rs

```diff
@@ -1,90 +1,316 @@
-pub use schema::*;
-use serde::{Deserialize, Serialize};
-use signature_bbs_plus::Signature;
-use signature_core::nonce::Nonce;
-pub use traits::*;
-pub use traits_credentials::*;
-use util::*;
-pub(crate) use workers::*;
-
-/// A credential that can be presented
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct BbsCredential {
-    /// The signed attributes in the credential
-    pub attributes: Vec<CredentialAttribute>,
-    /// The cryptographic signature
-    pub signature: Signature,
-}
-
-/// A credential and fragment 1 pair
-#[derive(Debug, Clone, Serialize, Deserialize)]
-pub struct CredentialRequestFragment(pub CredentialRequest, pub CredentialFragment1);
+#![allow(missing_docs)]
+
+mod identity;
+mod public_identity;
+mod storage_utils;
+mod worker;
+
+pub use storage_utils::*;
+
+use crate::IdentityIdentifier;
+use core::fmt;
+use core::marker::PhantomData;
+use core::time::Duration;
+use minicbor::bytes::ByteSlice;
+use minicbor::{Decode, Encode};
+use ockam_core::compat::borrow::Cow;
+use ockam_core::compat::{
+    collections::BTreeMap,
+    string::{String, ToString},
+    vec::Vec,
+};
+use ockam_core::{CowBytes, CowStr, Result};
+use serde::{Serialize, Serializer};
+
+#[cfg(feature = "tag")]
+use crate::TypeTag;
+
+pub const MAX_CREDENTIAL_VALIDITY: Duration = Duration::from_secs(30 * 24 * 3600);
+
+/// Type to represent data of verified credentials.
+#[derive(Debug, Encode)]
+pub enum Verified {}
+
+/// Type to represent data of unverified credentials.
+#[derive(Debug, Decode)]
+pub enum Unverified {}
+
+#[derive(Clone, Debug, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct Credential<'a> {
+    #[cfg(feature = "tag")]
+    #[n(0)] tag: TypeTag<3796735>,
+    /// CBOR-encoded [`CredentialData`].
+    #[b(1)] data: CowBytes<'a>,
+    /// Cryptographic signature of attributes data.
+    #[b(2)] signature: CowBytes<'a>,
+}
+
+impl fmt::Display for Credential<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.serialize(f)
+    }
+}
+
+#[derive(Debug, Decode, Encode)]
+#[rustfmt::skip]
+#[cbor(map)]
+pub struct CredentialData<'a, T> {
+    /// A schema identifier to allow distinguishing sets of attributes.
+    #[n(1)] schema: Option<SchemaId>,
+    /// User-defined key-value pairs.
+    #[b(2)] attributes: Attributes<'a>,
+    /// The subject this credential is issued for.
+    #[n(3)] subject: IdentityIdentifier,
+    /// The entity that signed this credential.
+    #[n(4)] issuer: IdentityIdentifier,
+    /// The label of the issuer's public key.
+    #[b(5)] issuer_key_label: CowStr<'a>,
+    /// The time when this credential was created.
+    #[n(6)] created: Timestamp,
+    /// The time this credential expires.
+    #[n(7)] expires: Timestamp,
+    /// Term to represent the verification status type.
+    #[n(8)] status: Option<PhantomData<T>>
+}
+
+impl<'a> CredentialData<'a, Unverified> {
+    pub(crate) fn make_verified(self) -> CredentialData<'a, Verified> {
```

Comment:
```suggestion
    pub(crate) fn into_verified(self) -> CredentialData<'a, Verified> {
```

Cf. https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv